python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1157,6 +1175,23 @@ def build_static_for(ctx, node, is_grouped):
         else:
             build_stmt(ctx, node.iter)
             targets = ASTTransformer.get_for_loop_targets(node)
+            iter_time = 0
+            if ti_unroll_limit:
+                for target_values in node.iter.ptr:
```

Comment:
Sorry but I've thought about it again and found that it can cause some problems if we iterate through a generator twice.
Consider this program:
```
import taichi as ti

ti.init()

a = filter(lambda x: x % 2 == 0, range(10))

@ti.kernel
def foo():
    for i in ti.static(a):
        print(i)

foo()
```
`filter` can only be iterated once, and it yields nothing on the second iteration, so the program will output nothing.